### PR TITLE
Luis fix add edit remove descriptions frontend

### PR DESCRIPTION
**Pull request by luisarevalo21**
# Description
Allow owners to edit a current warning description. 
Only owners are able to edit a warning description. 
Upon editing a warning the remaining warnings become uneditable until save is triggered.

## Related PRS (if any):
This frontend PR is related to the #[983](https://github.com/OneCommunityGlobal/HGNRest/pull/983) backend PR.

## Main changes explained:
- Inside of `WarningTrackerModal.jsx` I changed the p tag to input passing the styles and added the functionality needed.
- The functionality allows owners to edit by changing the same.
- Next a save button was added to trigger the actual save and put action sending the data to the backend.

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. log in as an owner account
4. Upon loading, click the "+/-" button to load all the current warnings. 
5. Upon loading, create a test, warning and submit it. 
6. Click the newly created warning, and edit it. 
7. Once edited, click the 'save' button to send the updated warning. 
8. View the user's warnings and should display the newly edited warning. Also, the `WarningTrackerModal.jsx`will have the updated list as well. 

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/fbbfc65f-10ca-492f-8a3c-52b4393cd3e5

## Note:
Include the information the reviewers need to know.
There is a bug, which when creating a new warning, it doesn't allow you to add spaces, I'll fix the issue in this PR #2037

### CODE REVIEW

File: src/actions/warnings.js [hunk 1]
```diff
@@ -61,3 +66,80 @@ export const deleteWarningsById = (warningId, personId) => {
...
+    } catch (error) {
+      return { error: error.message };
+    }
+  };
+};
+
+export const updateWarningDescription = warningDescriptionId => {
+  const url = ENDPOINTS.UPDATE_WARNING_DESCRIPTION(warningDescriptionId);
+
+  return async dispatch => {
```
Comment (TXMO-dev) [inline src/actions/warnings.js:135]: Great Job here as well but then I have a slight issue because of a potential race condition:
The function dispatch(updateWarningDescriptionAction(res.data)) returns a promise, and its result is then awaited. However, calling dispatch within another dispatch could lead to potential issues such as race conditions or unintended side effects if not handled carefully. Its crucial we look at this for predictable state management. Another issue i also noticed is, The return value of dispatch(updateWarningDescriptionAction(res.data)) is being awaited and then returned. It's important to make sure that this value is what you expect it to be.
Suggested Refactor:
```
export const updateWarningDescription = (warningDescriptionId) => {
  const url = ENDPOINTS.UPDATE_WARNING_DESCRIPTION(warningDescriptionId);

  return async (dispatch) => {
    try {
      // Perform the API call
      const res = await axios.put(url, { warningDescriptionId });
      
      // Dispatch the action and wait for its resolution
      const result = await dispatch(updateWarningDescriptionAction(res.data));

      // Ensure the response payload is returned correctly
      return result.payload;
    } catch (error) {
      // Log the error for debugging purposes
      console.error('Error updating warning description:', error);

      // Return a structured error object
      return { error: error.message || 'An error occurred' };
    }
  };
};
```

**Review by TXMO-dev — CHANGES_REQUESTED** [review]
Great Job, I left a review kindly take a look at it and let me know your feedback thank you.

**Review by JatinAgrawal94 — CHANGES_REQUESTED** [review]
New warning can be added. But it's not visible when I try to view it all. I guess it's due to the error visible in the console. Consider this video for more information.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47699187/d438b46a-5e1e-4d93-8beb-19ab5158fcc9

**Review by Kavil-Jain-514 — CHANGES_REQUESTED** [review]
A new Warning Tracker is not being added to the tracking list.
The video shows the reviewed PR.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112303944/a7084002-817f-42a7-b961-525ccefe61e0



Also, when an old warning is edited. This is the error I encounter
image

**Comment by luisarevalo21** [comment]
> New warning can be added. But it's not visible when I try to view it all. I guess it's due to the error visible in the console. Consider this video for more information.
> 
>  PR.2343.mp4

Hello @JatinAgrawal94 
Please pull latest changes to the backend, I forgot to add backend code to fetch the new warnings. 
As for the warning that appears, you can ignore it as it is irrelevant to this PR as it is error on teamMemberTask, but I didn't touch that file.
Thanks.

**Comment by luisarevalo21** [comment]
> A new Warning Tracker is not being added to the tracking list. The video shows the reviewed PR.
> 
>  PR.2343.mp4 
> Also, when an old warning is edited. This is the error I encounter image

Hello @Kavil-Jain-514 
I updated the backend as I forgot to add the code to fetch the updated warnings. 
Please pull and run the build command this should resolve the issue of warnings not updating and should fix the post method as well. 
Please re-review it thanks.

**Comment by akshit0211** [comment]
Hi, I am able to add a new warning and edit it. Good job!

warning1
added warning 1
warning2
added warning 2

**Review by akshit0211 — DISMISSED** [review]

**Review by JatinAgrawal94 — CHANGES_REQUESTED** [review]
1) New Warning can be added
2) Existing Warnings can be edited and saved.
3) Other warning becomes disabled while one is being saved after edit.

Issue: When I try to enter the warning name as "NW2" it gives me a message that warning already exists, though it doesn't.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47699187/27bac601-75cb-41e4-b550-e0d4a37dee18

**Review by Kavil-Jain-514 — COMMENTED** [review]
I am getting an error when editing a warning.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112303944/9de1792e-54e9-4024-aa42-f708f62afa1a

**Comment by luisarevalo21** [comment]
> I am getting an error when editing a warning.
> 
>  PR.2343.-.983.mp4

Hello 
Try fetching the backend, then do npm run build && npm run start. 
Let me know if it fixes it.

**Review by JatinAgrawal94 — DISMISSED** [review]
Changes work as expected.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47699187/907c7e08-b581-48a7-829b-9abd97bb0169

**Review by Kavil-Jain-514 — DISMISSED** [review]
The functionality is working as expected !!


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112303944/ef3dd54f-f068-4435-b085-a517e13cc1b8

**Review by Vigneshwar-Muriki — DISMISSED** [review]
Logged in as the owner and was able to add a new warning, edit it, and received an error message if the warning already exists. Great work!

https://github.com/user-attachments/assets/e4f32ba5-e175-403e-badb-0f4803d0bc0e

Screenshot 2024-07-13 212356

**Review by kabraambika — CHANGES_REQUESTED** [review]
I have verified the PR for both the frontend and related backend. The feature is working, but there's an error appearing in the console whenever there's a warning.

The console shows a 422 Unprocessable Entity error. Please refer to the attached video and screenshot for details.

https://github.com/user-attachments/assets/3600d0ae-63b5-4c67-943b-6baa759aaab0
image

**Review by SrikanthChowdary495 — COMMENTED** [review]
I am encountering an error. There are API errors.  I am experiencing an error with API. I have attached an image for reference.

**Comment by angelalalacheng** [comment]
Hi,
I encountered the API error as well.

https://www.loom.com/share/3191e0ecddc84d14a19aea9cdfa0c4ac?sid=07371a25-7c6e-4ff5-9fec-0f2111f5c20b

**Review by varunreddy03 — DISMISSED** [review]
I have checked into both frontend and backend for  this PR and have tested the feature and it is working as expected.

PR 2343_1
PR 2343_2
PR 2343_3

**Comment by ReinaT5678** [comment]
I'm also experiencing an API error:

This pops up when I try to open the "+/-" button under any user:

Screenshot 2024-08-02 at 7 34 59 PM

This is the error when I try to enter a warning tracker title:
Screenshot 2024-08-02 at 7 33 48 PM

**Comment by luisarevalo21** [comment]
> Hello
> Try fetching the backend, then do npm run build && npm run start.
> Let me know if it fixes it.

Hello 

Please check into the backend, then do inside of the folder run the following command `npm run build && npm run start`
Let me know if it fixes it.

**Comment by luisarevalo21** [comment]
> I'm also experiencing an API error:
> 
> This pops up when I try to open the "+/-" button under any user:
> 
> Screenshot 2024-08-02 at 7 34 59 PM
> This is the error when I try to enter a warning tracker title: Screenshot 2024-08-02 at 7 33 48 PM

Hello @ReinaT5678 

Please check into the backend, then do inside of the folder run the following command npm run build && npm run start
Let me know if it fixes it.

**Comment by luisarevalo21** [comment]
> Hi, I encountered the API error as well.
> 
> https://www.loom.com/share/3191e0ecddc84d14a19aea9cdfa0c4ac?sid=07371a25-7c6e-4ff5-9fec-0f2111f5c20b


Hello @angelalalacheng 

Please check into the backend, then do inside of the folder run the following command npm run build && npm run start
Let me know if it fixes it.

**Comment by luisarevalo21** [comment]
Hello @kabraambika 
This error is coming from the warning.js file on line 34, when you are attempting to add a duplicate warning. 
I tried debugging it to handle it correctly so it doesn't appear in the console and just display the error message on the modal but was unsuccessful do you have any ideas on how to solve this problem? 
Thanks

**Comment by kabraambika** [comment]
> Hello @kabraambika This error is coming from the warning.js file on line 34, when you are attempting to add a duplicate warning. I tried debugging it to handle it correctly so it doesn't appear in the console and just display the error message on the modal but was unsuccessful do you have any ideas on how to solve this problem? Thanks

Hello @luisarevalo21, I reviewed the implementation and noticed that you're using a 422 status code for duplicates, which results in the console error. If we are continuing with correct implementation then console errors will appear whenever 422 status code will be sent. 
As a suggestion, instead of sending a 422 status for warnings like the presence of special characters or duplicate entries, you could use a 200 status code with a success boolean set to false.

Changes in backend current code
image

Changes in frontend current code
image

**Review by Pranay-ai — DISMISSED** [review]
Hey @luisarevalo21 
The PR is working as expected here is the screenshot for reference 
image

The Modal is having the Latest Updates as well.
Thank YOU

**Review by Parth-tech — DISMISSED** [review]
Hey Luis,
The PR meets its required functionality and the codebase looks good too. Great job!

Image References:
Screenshot 2024-08-06 at 5 14 05 PM
Screenshot 2024-08-06 at 5 14 19 PM
Screenshot 2024-08-06 at 5 14 38 PM
Screenshot 2024-08-06 at 5 15 00 PM

**Comment by arunsrao9** [comment]
I am getting this error even after I run the command npm run build && npm run start
Screenshot 2024-08-06 at 6 13 05 PM
Screenshot 2024-08-06 at 6 12 59 PM

**Comment by luisarevalo21** [comment]
> > Hello @kabraambika This error is coming from the warning.js file on line 34, when you are attempting to add a duplicate warning. I tried debugging it to handle it correctly so it doesn't appear in the console and just display the error message on the modal but was unsuccessful do you have any ideas on how to solve this problem? Thanks
> 
> Hello @luisarevalo21, I reviewed the implementation and noticed that you're using a 422 status code for duplicates, which results in the console error. If we are continuing with your implementation then console errors will appear whenever 422 status code will be sent. As a suggestion, instead of sending a 422 status for warnings like the presence of special characters or duplicate entries, you could use a 200 status code with a success boolean set to false.
> 
> Changes in backend current code image
> 
> Changes in frontend current code image

Hello @kabraambika 
I used a 422 as I wanted it to send an error back since an error occurred when attempting to create duplicate names or use special characters. I saw 200 worked when I was debugging. Do you know of a way to send the 422 error and not get the error message on the console? 
Otherwise then yes I'll just send back 200 and set a flag saying duplicates aren't allowed.

**Comment by kabraambika** [comment]
> > > Hello @kabraambika This error is coming from the warning.js file on line 34, when you are attempting to add a duplicate warning. I tried debugging it to handle it correctly so it doesn't appear in the console and just display the error message on the modal but was unsuccessful do you have any ideas on how to solve this problem? Thanks
> > 
> > 
> > Hello @luisarevalo21, I reviewed the implementation and noticed that you're using a 422 status code for duplicates, which results in the console error. If we are continuing with your implementation then console errors will appear whenever 422 status code will be sent. As a suggestion, instead of sending a 422 status for warnings like the presence of special characters or duplicate entries, you could use a 200 status code with a success boolean set to false.
> > Changes in backend current code image
> > Changes in frontend current code image
> 
> Hello @kabraambika I used a 422 as I wanted it to send an error back since an error occurred when attempting to create duplicate names or use special characters. I saw 200 worked when I was debugging. Do you know of a way to send the 422 error and not get the error message on the console? Otherwise then yes I'll just send back 200 and set a flag saying duplicates aren't allowed.

Hi @luisarevalo21 , I'm not aware of any method to send a 422 status from the backend while whitelisting it for the browser console.

**Review by AnkitLall — CHANGES_REQUESTED** [review]
I have tested the feature, following the instructions in the PR, and the update feature works and duplicate entry error works too. Please refer the screen recording attached below.

Although, I found a small discrepancy. The warning text in the modal gets changed to Title Case while the text in the warnings list remains unchanged. Refer the attached screenshot.


https://github.com/user-attachments/assets/ee13b6d5-9529-4466-a3ba-f94c49a7fdff

Screen Shot 2024-08-10 at 2 57 58 PM

**Review by adityasure9 — DISMISSED** [review]
The functionality to add, edit, and remove warnings works as expected. The ability for owners to manage warnings has been implemented well, and the updates reflect correctly in the UI.

**Comment by luisarevalo21** [comment]
> I have tested the feature, following the instructions in the PR, and the update feature works and duplicate entry error works too. Please refer the screen recording attached below.
> 
> Although, I found a small discrepancy. The warning text in the modal gets changed to Title Case while the text in the warnings list remains unchanged. Refer the attached screenshot.
> 
>  Screen.Recording.2024-08-10.at.2.29.09.PM.mp4 
> Screen Shot 2024-08-10 at 2 57 58 PM

Hello @AnkitLall 
That was the intended functionality, was to store, the warnings as lowercased and CSS was used to uppercase the warning. 
However, I can see the argument as whatever the owner types should be stored. My issue would be the CSS will capitalize on the modal I guess I can remove it. What are your thoughts? I'd like to hear them.

**Comment by luisarevalo21** [comment]
> > > > Hello @kabraambika This error is coming from the warning.js file on line 34, when you are attempting to add a duplicate warning. I tried debugging it to handle it correctly so it doesn't appear in the console and just display the error message on the modal but was unsuccessful do you have any ideas on how to solve this problem? Thanks
> > > 
> > > 
> > > Hello @luisarevalo21, I reviewed the implementation and noticed that you're using a 422 status code for duplicates, which results in the console error. If we are continuing with your implementation then console errors will appear whenever 422 status code will be sent. As a suggestion, instead of sending a 422 status for warnings like the presence of special characters or duplicate entries, you could use a 200 status code with a success boolean set to false.
> > > Changes in backend current code image
> > > Changes in frontend current code image
> > 
> > 
> > Hello @kabraambika I used a 422 as I wanted it to send an error back since an error occurred when attempting to create duplicate names or use special characters. I saw 200 worked when I was debugging. Do you know of a way to send the 422 error and not get the error message on the console? Otherwise then yes I'll just send back 200 and set a flag saying duplicates aren't allowed.
> 
> Hi @luisarevalo21 , I'm not aware of any method to send a 422 status from the backend while whitelisting it for the browser console.

Hello @kabraambika 
Thanks for the response, I'll go ahead and use a 200, it will remove the 422 on the console. I'll ping you once I push the changes. 
Thanks!

**Comment by luisarevalo21** [comment]
> > > > > Hello @kabraambika This error is coming from the warning.js file on line 34, when you are attempting to add a duplicate warning. I tried debugging it to handle it correctly so it doesn't appear in the console and just display the error message on the modal but was unsuccessful do you have any ideas on how to solve this problem? Thanks
> > > > 
> > > > 
> > > > Hello @luisarevalo21, I reviewed the implementation and noticed that you're using a 422 status code for duplicates, which results in the console error. If we are continuing with your implementation then console errors will appear whenever 422 status code will be sent. As a suggestion, instead of sending a 422 status for warnings like the presence of special characters or duplicate entries, you could use a 200 status code with a success boolean set to false.
> > > > Changes in backend current code image
> > > > Changes in frontend current code image
> > > 
> > > 
> > > Hello @kabraambika I used a 422 as I wanted it to send an error back since an error occurred when attempting to create duplicate names or use special characters. I saw 200 worked when I was debugging. Do you know of a way to send the 422 error and not get the error message on the console? Otherwise then yes I'll just send back 200 and set a flag saying duplicates aren't allowed.
> > 
> > 
> > Hi @luisarevalo21 , I'm not aware of any method to send a 422 status from the backend while whitelisting it for the browser console.
> 
> Hello @kabraambika Thanks for the response, I'll go ahead and use a 200, it will remove the 422 on the console. I'll ping you once I push the changes. Thanks!

Hello @kabraambika 
I adjusted the 422 error and no longer returns a post method on the frontend, and changed it to a 200.

**Review by SMitta11 — DISMISSED** [review]
I login as owner and can see +-below users name after running frontend and backend:
Screen Shot 2024-08-17 at 5 07 18 PM
Added a new warning
Screen Shot 2024-08-17 at 5 07 57 PM
Then clicked again on tracking and can see newly added warning:
Screen Shot 2024-08-17 at 5 08 38 PM
As given in PR, edited this new warning and saved it:
Screen Shot 2024-08-17 at 5 11 43 PM
Then clicked on tracking and can see this new warning
Screen Shot 2024-08-17 at 5 12 15 PM
Checked that this warning is appearing in front of other user's name:
Screen Shot 2024-08-17 at 5 12 26 PM
Also checked the console:
Screen Shot 2024-08-17 at 5 16 05 PM

**Review by Lemonnycodes — DISMISSED** [review]
Hi,

I tested the PR, including creating, editing, and displaying warnings. Everything works as expected.

Reference Images:
PR 2343 (1)
PR 2343 (2)
PR 2343 (3)
PR 2343 (4)
PR 2343 (5)

**Review by yashwanth170 — DISMISSED** [review]
The functionality works as expected. Attaching screenshots.
test_demo_1
test_demo_2
test_demo_3

**Review by AJAYINAVOLU — DISMISSED** [review]
Changes are working as expected. 

PR 2343
PR 2343-1

Thank You

**Review by ishangoel17 — COMMENTED** [review]
Verified the PR and the changes are working as expected. Able to add new warning and view/edit it. However, on editing and saving the edited warning, a space appears on the top of the warnings list as shown in screenshot5. Sometimes for a short second there is message saying the warning edit has been saved successfully but its intermittent and does not appear every time on editing a warning. On the times, the message does not appear, there is just the empty space as shown in screenshot5. Below are the screenshots for reference:
Capture2343
Capture2343_2
Capture2343_3
Capture2343_4
Capture2343_5

**Review by Shreyav2000 — CHANGES_REQUESTED** [review]
The warnings are being added and deleted properly as intended, but the system is allowing duplicate warnings to be created, which it shouldn’t. This issue arises despite the presence of the checkForDuplicates function, which is designed to prevent this behavior. The problem likely stems from the function not being executed at the correct point in the process or not properly comparing warning titles. It could be failing due to inconsistencies like case sensitivity, spacing, or trimming that prevent it from detecting duplicates effectively. I am attaching a video below showing the same and a screenshot for more clarification. Thank you!

https://github.com/user-attachments/assets/7d4ec9f9-2eda-4854-83a9-d4ccf6f008b5

Screenshot 2024-09-06 at 12 40 35 AM

**Review by vedantg24 — DISMISSED** [review]
Tested the PR, and it is working as expected. Warnings can be added and edited successfully. Also, tried to add a duplicate warning, and it displayed the error for “warning already exists”.

Screenshot 2024-09-06
Screenshot 2024-09-06
Screenshot 2024-09-06
Screenshot 2024-09-06

**Comment by luisarevalo21** [comment]
> The warnings are being added and deleted properly as intended, but the system is allowing duplicate warnings to be created, which it shouldn’t. This issue arises despite the presence of the checkForDuplicates function, which is designed to prevent this behavior. The problem likely stems from the function not being executed at the correct point in the process or not properly comparing warning titles. It could be failing due to inconsistencies like case sensitivity, spacing, or trimming that prevent it from detecting duplicates effectively. I am attaching a video below showing the same and a screenshot for more clarification. Thank you!
> 
>  Screen.Recording.2024-09-06.at.12.41.12.AM.mov 
> Screenshot 2024-09-06 at 12 40 35 AM

Hello @Shreyav2000 
Please pull the most recent code from this branch. 
Run the the command 'git pull' while on this branch and should fetch the most updated code. 
I tried to duplicate your error, but shows I cannot have duplicates. 
Also, when running the backend pull as well, and run the command 'npm run build && npm run start' 
I've attached a video of my example. 
Thanks! 

https://github.com/user-attachments/assets/950496a9-fe20-4e9c-973e-2525ec8d9ffd

**Comment by saurabhshetty29** [comment]
Encountered the following error when clicking on the '+/-' button:
Screenshot 2024-09-10 at 11 31 16 PM

An error occurs when attempting to add a warning:
Screenshot 2024-09-10 at 11 31 44 PM

**Comment by luisarevalo21** [comment]
> Encountered the following error when clicking on the '+/-' button: Screenshot 2024-09-10 at 11 31 16 PM
> 
> An error occurs when attempting to add a warning: Screenshot 2024-09-10 at 11 31 44 PM

@saurabhshetty29 please ensure you've pulled the latest changes. 
also make sure you pulled the appropriate backend as well as stated in the post. 
If you still have issues feel free to reach out.

**Review by crystallow1168 — DISMISSED** [review]
Hi @luisarevalo21,

Changes looks good and working as expected. Below are my findings:

https://github.com/user-attachments/assets/633fb9bd-7f5c-4497-94f2-81fdb0b5ce2d


✅ Owner user can create a warning and warning can be seen when clicked on the "Tracking" button.
✅ Owner user can edit the warning and changes can be seen under the tracking warning section

**Comment by sud306** [comment]
Hello,
I reviewed PR983 and PR2343 and found that it's working as expected.

9 PR983_PR2343_1

9 PR983_PR2343_2

**Review by SFA23SCM35V — DISMISSED** [review]
I have reviewed both frontend and backend PR's and no defect found 
Great job Approving the change

**Review by neeharikakoniki — APPROVED** [review]
The warnings creation and editing functionality works smoothly. The UI updates immediately, and the WarningTrackerModal.jsx reflects the changes accurately.

image
image

**Review by ashayk9 — COMMENTED** [review]
I used the proper frontend and backend branches as per the instructions.
But encountered the following error when clicking on the '+/-' button:
Screenshot 2024-09-19 at 12 31 52 PM
Screenshot 2024-09-19 at 12 32 54 PM
Let me know if I need to test it in a different way

**Review by Muzammil603 — CHANGES_REQUESTED** [review]
image
I am not getting +/- option

**Comment by pallavithorat** [comment]
I tested the PR and encountered an issue where the "Add New Warning Tracker" functionality is returning an error message, "error occurred" (screenshot attached).
ss
Screenshot 2024-10-06 at 8 18 37 PM

Please let me know if there's a different way I should be testing this.

**Comment by luisarevalo21** [comment]
> I tested the PR and encountered an issue where the "Add New Warning Tracker" functionality is returning an error message, "error occurred" (screenshot attached). ss Screenshot 2024-10-06 at 8 18 37 PM
> 
> Please let me know if there's a different way I should be testing this.

Hello 
Please ensure the frontend is running node 14.17.6. 
Also, follow the backend and run npm run build && npm run start to run and build the backend server. 
Reach out to me if you need further assistance.

**Comment by luisarevalo21** [comment]
> image I am not getting +/- option

Please, read the guidelines. 
You are most likely not logged in as an owner account hence why you cannot see the +/- button. 
Feel free to reach out if you need further assistance.

**Comment by luisarevalo21** [comment]
> Hello
> Please ensure the frontend is running node 14.17.6.
> Also, follow the backend and run npm run build && npm run start to run and build the backend server.
> Reach out to me if you need further assistance.

Hello
Please ensure the frontend is running node 14.17.6.
Also, follow the backend and run npm run build && npm run start to run and build the backend server.
Reach out to me if you need further assistance.

**Review by samarth9201 — APPROVED** [review]
Verified the PR. LGTM!

Screenshot (113)
Screenshot (114)
Screenshot (115)
Screenshot (116)

File: src/components/Warnings/WarningsModal.jsx [hunk 2]
```diff
@@ -1,106 +1,106 @@
-/* eslint-disable no-unused-vars */
-/* eslint-disable spaced-comment */
-/* eslint-disable no-alert */
-/* eslint-disable no-undef */
-/* eslint-disable react/no-unescaped-entities */
-/* eslint-disable react/destructuring-assignment */
-import { Modal, ModalHeader, ModalBody, ModalFooter, Button, Row } from 'reactstrap';
+// /* eslint-disable no-unused-vars */
```
Comment (sheetalmangate) [inline src/components/Warnings/WarningsModal.jsx:1]: Instead of commenting out all the code in the file, it would be better to delete the file.

**Review by Supriya1659 — APPROVED** [review]
Initially, I encountered an error, but after referring to the comments in the backend and executing npm run build and npm run start, I observed the warnings. I was also able to add and edit my own warnings, and everything is now working as expected.


image

image

image

image

image

**Review by ajitp4292 — APPROVED** [review]
PR 2343 is working as expected. When I add a new warning tracker, I see that the tracking and warning tracker can be edited. Once edited, it displays the correct name.
Screenshot 2024-11-09 at 7 26 58 PM
Screenshot 2024-11-09 at 7 30 08 PM
Screenshot 2024-11-09 at 7 30 32 PM
Screenshot 2024-11-09 at 7 30 51 PM

**Review by Sheehan20 — COMMENTED** [review]
Hi @luisarevalo21, great work on implementing the warning description management feature! Here's a detailed review:

1. Feature Highlights:
- Added functionality for owners to edit warning descriptions
- Implemented restrictions to prevent multiple simultaneous edits
- Created a new modal for warning tracker management

2. Key Improvements:
- Enhanced warning management workflow
- Added save mechanism for edited warnings
- Implemented owner-specific access controls

3. Technical Observations:
- Added new actions and reducers for warning description management
- Created modular components for warning interactions
- Implemented error handling for various warning-related operations

4. Potential Considerations:
- The PR mentions a bug with adding spaces to new warnings (to be addressed in PR #2037)
- Some reviewers noted occasional issues with warning creation and editing

5. Testing Recommendations:
- Verify warning creation across different scenarios
- Test edge cases for warning editing
- Ensure proper access control for warning management

Additional Notes:
- Recommended addressing the potential race condition in state management (as noted by @TXMO-dev)
- Consider adding more robust error handling

Overall, this is a solid implementation that improves the warning tracking functionality. Great job on the detailed implementation!

**Review by Srichand-Medagani — APPROVED** [review]
Works perfectly! I have left my comments and video on backend.

**Review by SSP24SCM16G — APPROVED** [review]
Everything is working good.
Screenshot 2025-03-01 at 19 47 35
Screenshot 2025-03-01 at 19 47 50
Screenshot 2025-03-01 at 19 50 10
Screenshot 2025-03-01 at 19 50 28

**Review by smohansatyaram — COMMENTED** [review]
Owner edit permissions work as expected
Warning description updates correctly
Save functionality properly triggers backend updates

Note on Dark Mode Visibility:
In dark mode, the black text on dark backgrounds has low contrast, which may make it challenging for some users to read. (See last image)

Screenshot 2025-03-27 192912
Screenshot 2025-03-27 192927
Screenshot 2025-03-27 192941
Screenshot 2025-03-27 193035

**Review by karnati007 — APPROVED** [review]
Tested the Warning Tracker feature by logging in as an owner account. Verified that clicking the "+/-" button correctly loads all current warnings. Successfully created a test warning, submitted it, and confirmed its appearance in the list. Edited the warning, saved changes, and verified that the updated warning reflected correctly in the user's warnings section and in the WarningTrackerModal.jsx. Cleared site data before testing. Everything is functioning as expected. 
LuisPR2
LuisPR

**Review by sawantgayatri19 — CHANGES_REQUESTED** [review]
Hi Luis,

Great work implementing the frontend support for adding, editing, and removing warning descriptions! I've reviewed the code changes and verified the workflow. 

Core Functionality:Logged in as an owner, clicking the `+/-` button correctly opens the tracking modal to add, edit, and save warning descriptions.
State & UI Updates: The transition from text elements to editable input fields within [WarningTrackerModal.jsx](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2343) handles updates cleanly.
Action Items Before Merge:
1. Merge Conflicts:Please resolve the active merge conflicts across the branch files (including `src/actions/warnings.js`, `src/components/Warnings/Warnings.jsx`, and `src/components/Warnings/modals/WarningTrackerModal.jsx`).
2. Async Dispatch Handling:Consider refactoring the nested dispatch calls in `src/actions/warnings.js` to mitigate potential race conditions, as flagged by earlier reviewers.
3. Code Cleanup: Ensure large blocks of commented-out code (e.g., in `WarningsModal.jsx`) are completely removed rather than left commented out.
4. Dark Mode Polish:Verify text contrast on warning modals under dark mode to ensure clear visibility.

Once these items and conflicts are cleared up, this will be ready to merge!
